python/cuml/test/test_metrics.py

```diff
@@ -152,3 +154,96 @@ def test_rand_index_score(name, nrows):
     cu_score_using_sk = sk_ars(y, cu_y_pred)
 
     assert array_equal(cu_score, cu_score_using_sk)
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_homogeneity_score(use_handle):
+    def score_labeling(ground_truth, predictions):
+        a = np.array(ground_truth, dtype=np.int)
+        b = np.array(predictions, dtype=np.int)
+
+        a_dev = cuda.to_device(a)
+        b_dev = cuda.to_device(b)
+
+        handle, stream = get_handle(use_handle)
+
+        return cuml.metrics.homogeneity_score(a_dev, b_dev, handle=handle)
+
+    # Perfect labelings are homogeneous
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [1, 1, 0, 0]),
+                                   1.0, decimal=7)
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 0, 1, 1]),
+                                   1.0, decimal=7)
+
+    # Non-perfect labelings that further split classes into more clusters can
+    # be perfectly homogeneous
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 0, 1, 2]),
+                                   1.0, decimal=7)
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 1, 2, 3]),
+                                   1.0, decimal=7)
+
+    # Clusters that include samples from different classes do not make for an
+    # homogeneous labeling
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 1, 0, 1]),
+                                   0.0, decimal=7)
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 0, 0, 0]),
+                                   0.0, decimal=7)
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_homogeneity_score_big_array(use_handle):
+    def assert_ours_equal_sklearn(random_generation_lambda):
+        rng = np.random.RandomState(1234)  # makes it reproducible
+        a = random_generation_lambda(rng)
+        b = random_generation_lambda(rng)
+
+        a_dev = cuda.to_device(a)
+        b_dev = cuda.to_device(b)
+
+        handle, stream = get_handle(use_handle)
+
+        score = cuml.metrics.homogeneity_score(a_dev, b_dev, handle=handle)
+        ref = sk_hom_score(a_dev, b_dev)
+
+        np.testing.assert_almost_equal(score, ref, decimal=7)
+
+    assert_ours_equal_sklearn(lambda rng: rng.randint(0, 1000, int(10e4)))
+    assert_ours_equal_sklearn(lambda rng: rng.randint(-1000, 1000, int(10e4)))
+
+
+def assert_mi_equal_sklearn(a, b, use_handle):
+    a_dev = cuda.to_device(a)
+    b_dev = cuda.to_device(b)
+
+    handle, stream = get_handle(use_handle)
+
+    score = cuml.metrics.mutual_info_score(a_dev, b_dev, handle=handle)
+    ref = sk_mi_score(a_dev, b_dev)
+    np.testing.assert_almost_equal(score, ref, decimal=7)
```

Comment:
That is a pretty tight bound with decimal=7... we could probably survive with decimal=4 or something similar to prevent test failures due to very tiny differences in score

Reply:
Agreed.